modules/core/shared/src/main/scala/ciris/ConfigValue.scala

```diff
@@ -293,24 +266,28 @@ sealed abstract class ConfigValue[+F[_], A] {
     * `.redacted.map(Secret(_))`.
     */
   final def secret(implicit show: Show[A]): ConfigValue[F, Secret[A]] =
-    transform {
+    itransform({
       case Default(error, a)     => Default(error.redacted, Secret(a)(show))
       case Failed(error)         => Failed(error.redacted)
       case Loaded(error, key, a) => Loaded(error.redacted, key, Secret(a)(show))
-    }
+    })(_.value)
+
+  final def fields: List[ConfigField] =
```

Comment:
Do you think we should introduce a `ConfigFields` type or similar? So we can provide some rendering methods (and control `toString`) directly in the returned type?

Reply:
I'm not sure about the relevance of a `ConfigFieldList` or `ConfigFields` wrapper :thinking:. What would be the advantage over `List[ConfigField]` with function taking it as a parameter e.g `renderMarkdown(fields: List[ConfigField])`?

Reply:
I guess it has better discoverability (e.g. `config.fields.renderMarkdown` vs. discovering the `renderMarkdown` function). And as we add more renderers (or other functionality), it's convenient to find them all through autocomplete on `fields.`. Plus we can control `ConfigFields#toString` and do something better than `List#toString`.